app/tests/tests_10/test_external_collection_assets_endpoint.py

```diff
@@ -0,0 +1,373 @@
+from tests.utils import mock_s3_asset_file
+
+logger = logging.getLogger(__name__)
+
+
+@override_settings(FEATURE_AUTH_ENABLE_APIGW=True)
+class CollectionAssetsExternalAssetEndpointTestCase(StacBaseTestCase):
+
+    @mock_s3_asset_file
+    def setUp(self):  # pylint: disable=invalid-name
+        self.factory = Factory()
+        self.collection = self.factory.create_collection_sample().model
+        self.client = Client(headers=get_auth_headers())
+        self.maxDiff = None  # pylint: disable=invalid-name
+
+    def test_create_collection_asset_with_external_url(self):
+        collection = self.collection
+        collection_asset_data = {
+            'id': 'clouds.jpg',
+            'description': 'High in the sky',
+            'type': 'image/jpeg',  # specify the file explicitly
+            'href': settings.EXTERNAL_TEST_ASSET_URL
+        }
+
+        # create the asset, which isn't allowed
+        response = self.client.put(
+            reverse_version(
+                'collection-asset-detail', args=[collection.name, collection_asset_data['id']]
+            ),
+            data=collection_asset_data,
+            content_type="application/json"
+        )
+
+        self.assertStatusCode(400, response)
+
+        collection.allow_external_assets = True
+        collection.external_asset_whitelist = [settings.EXTERNAL_TEST_ASSET_URL]
+        collection.save()
+
+        # create the asset, now it's allowed
+        response = self.client.put(
+            reverse_version(
+                'collection-asset-detail', args=[collection.name, collection_asset_data['id']]
+            ),
+            data=collection_asset_data,
+            content_type="application/json"
+        )
+
+        json_data = response.json()
+        self.assertStatusCode(201, response)
+        self.assertEqual(json_data['href'], collection_asset_data['href'])
+
+        new_collection_asset = CollectionAsset.objects.last()
+        self.assertEqual(new_collection_asset.file, collection_asset_data['href'])
+        self.assertTrue(new_collection_asset.is_external)
+        logger.debug("created asset file size %s", new_collection_asset.file_size)
+        self.assertEqual(new_collection_asset.file_size, 0)
+
+    @responses.activate
+    def test_create_collection_asset_validate_external_url(self):
+        collection = self.collection
+        external_test_asset_url = 'https://example.com/api/123.jpeg'
+        collection.allow_external_assets = True
+        collection.external_asset_whitelist = ['https://example.com']
+        collection.save()
+
+        # Mock response of external asset url
+        responses.add(
+            method=responses.GET,
+            url=external_test_asset_url,
+            body='som',
+            status=200,
+            content_type='application/json',
+            adding_headers={'Content-Length': '3'},
+            match=[responses.matchers.header_matcher({"Range": "bytes=0-2"})]
+        )
```

Comment:
Is the mocking is really needed for the call bellow to work ? Why ? Maybe a comment would be good here